src/services/products.js

```diff
@@ -0,0 +1,108 @@
+const { Product } = require('../models');
+const { productImageService } = require('../uploads');
+
+class ProductsService {
+  async createProduct (req, res) {
+    await productImageService.uploadImage(req);
+    const newProduct = await new Product(req.body).save();
+
+    return res.json(newProduct);
+  }
+
+  async updateProduct (req, res) {
+    if (req.files) {
+      const checkImage = await Product.findById(req.params.id).exec();
+
+      if (checkImage.image.length !== 0) {
+        await productImageService.deleteImage(checkImage.image);
+      }
+
+      await productImageService.uploadImage(req);
+    }
+
+    const updateProduct = await Product.findByIdAndUpdate(
+      req.params.id,
+      req.body
+    ).exec();
+
+    return updateProduct ? res.json(true) : res.boom.notFound();
+  }
+
+  async getProducts (req, res) {
+    const { _id, name, category, sizes, status, minPrice, maxPrice } =
+            (typeof req.query !== 'undefined' && req.query) || {};
+    const query = {};
+    const regexNums = /^[0-9.]*$/gm;
+    const regexForId = /^[0-9a-zA-Z]*$/gm;
+    const lengthForId = 24;
+
+    if (
+      _id != null &&
+            _id.match(regexForId) &&
+            _id.length === lengthForId
+    ) {
+      query._id = _id;
+    }
+    if (name != null) query.name = name;
```

Comment:
Name must be regex. There will be search on client part, so we must be able to find product not only by full name